pkg/client/apiutil/dynamicrestmapper.go

```diff
@@ -31,6 +31,9 @@ import (
 
 // ErrRateLimited is returned by a RESTMapper method if the number of API
 // calls has exceeded a limit within a certain time period.
+// Deprecated: as the dynamic RESTMapper now returns NoMatch errors directly
```

Comment:
```suggestion
// calls has exceeded a limit within a certain time period.
//
// DEPRECATED: As the dynamic RESTMapper now returns NoMatch errors directly
```

Just to give more emphasis :)

Reply:
Given that we're working on v0.7.x on the main branch, can we actually just remove this to avoid confusion? 

cc @alvaroaleman @DirectXMan12

Reply:
> Just to give more emphasis :)

`Deprecated:` is actually convention and some tools test for this explicitly (ref https://github.com/golang/go/wiki/Deprecated), so I would keep it, if you don't mind :)

Reply:
> Given that we're working on v0.7.x on the main branch, can we actually just remove this to avoid confusion?

Sure, I'm open for it. You decide, what better fits you :)

Reply:
> Deprecated: is actually convention and some tools test for this explicitly (ref https://github.com/golang/go/wiki/Deprecated), so I would keep it, if you don't mind :)

SGTM, let's add a space though between the deprecation notice and the original comment, if we're going down this path.

> Sure, I'm open for it. You decide, what better fits you :)

I'm +1 for it, although I'd like to wait to hear from other folks

Reply:
One additional option would be to keep this PR backwards-compatible as it is, so we can cherry-pick it to `0.6` and `0.5`, and remove the deprecated types/funcs in a separate PR only targeting `>= 0.7`.

Reply:
Hm, not sure if we want to backport these changes, given that's a breaking behavior change (check one error vs another) and could impact some users

Reply:
Yeah, fair enough...

Reply:
This PR is never going to be backwards compatible even if we keep the error type because that error type is never returned (which is btw much more subtile and annoying to debug than a compilation error). So I am very much in favor of just removing it.